thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/ThriftCallServiceBuilder.java

```diff
@@ -0,0 +1,127 @@
+ *                  // Adds multiple services by map
+ *                 .addServices(ImmutableMap.of("fooIterableMap",
+ *                                              ImmutableList.of(fooServiceImpl, barServiceImpl)))
+ *                 .build();
+ *  * }</pre>
+ *
+ * @see ThriftCallService
+ */
+public final class ThriftCallServiceBuilder {
+    private final ImmutableListMultimap.Builder<String, Object> servicesBuilder =
+            ImmutableListMultimap.builder();
+
+    private boolean useBlockingTaskExecutor;
+
+    ThriftCallServiceBuilder() {}
+
+    /**
+     * Adds multiple services by map for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addServices(Map<String, ?> services) {
+        requireNonNull(services, "services");
+        if (services.isEmpty()) {
+            throw new IllegalArgumentException("empty services");
+        }
+        services.forEach((k, v) -> {
+            if (v instanceof Iterable<?>) {
+                servicesBuilder.putAll(k, (Iterable<?>) v);
+            } else {
+                servicesBuilder.put(k, v);
+            }
+        });
+        return this;
+    }
+
+    /**
+     * Adds an service for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addService(Object... service) {
+        requireNonNull(service, "service");
+        return addService("", service);
+    }
+
+    /**
+     * Adds an service with a key for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addService(String key, Object... service) {
+        requireNonNull(service, "service");
+        servicesBuilder.putAll(key, service);
+        return this;
+    }
+
+    /**
+     * Adds services with key by iterable for {@link ThriftServiceEntry}.
+     */
+    public ThriftCallServiceBuilder addService(String key, Iterable<?> service) {
+        if (!service.iterator().hasNext()) {
+            throw new IllegalArgumentException("service should not be empty");
+        }
+        servicesBuilder.putAll(key, service);
+        return this;
+    }
+
+    /**
+     * Sets whether the service executes service methods using the blocking executor. By default, service
+     * methods are executed directly on the event loop for implementing fully asynchronous services. If your
+     * service uses blocking logic, you should either execute such logic in a separate thread using something
+     * like {@link Executors#newCachedThreadPool()} or enable this setting.
+     */
+    public ThriftCallServiceBuilder useBlockingTaskExecutor(boolean useBlockingTaskExecutor) {
+        this.useBlockingTaskExecutor = useBlockingTaskExecutor;
+        return this;
+    }
+
+    /**
+     * Builds a new instance of {@link ThriftCallService}.
+     */
+    public ThriftCallService build() {
+        return new ThriftCallService(
+                Multimaps.asMap(servicesBuilder.build()).entrySet().stream().collect(
```

Comment:
Think we can just call `asMap`
```suggestion
                servicesBuilder.build().asMap().entrySet().stream().collect(
```

Reply:
Done! Thank you!